storage/innobase/include/dict0mem.h

```diff
@@ -1543,6 +1543,60 @@ struct dict_vcol_templ_t {
 	dict_vcol_templ_t() : vtempl(0), mysql_table_query_id(~0ULL) {}
 };
 
+
+struct col_map_element_t
+{
+	static constexpr uint16_t DROPPED = 1U << 15;
+	static constexpr uint16_t NOT_NULL = 1U << 14;
+	static constexpr uint16_t UNSIGNED_SIZE = 0xfU << 10;
+	static constexpr uint16_t IND = 0x3ffU;
+	uint16_t data;
+	bool dropped() const
+	{
+		return data & DROPPED;
+	}
```

Comment:
There should be a `/** ` comment before the `struct`, and the type name should start with `field_map`. I would even declare the type nested inside `dict_instant_t`, because it should not be used elsewhere. `dict_instant_t::field_map_t` perhaps?

I think that there should be a `constexpr` or a `#define` for the 10 as well, and the various `:10` in InnoDB data structures should use a single constant. In any case, `IND` must be defined like `(1U << 10) - 1`, to clearly document the relationship with `UNSIGNED_SIZE`.

Do we really need all 4 bits for `UNSIGNED_SIZE`? This field should not yet be merged to 10.4; it is part of MDEV-15563. Do we need other values than 0,1,2,4? The maximum width of an integer type ought to be 8 bytes, and a 8-byte integer cannot be instantly converted from unsigned to signed. 2 bits should be enough for representing the values 0,1,2,4.

Each `constexpr` should be declared `private` to enforce the use of the accessor functions. And I would appreciate `/**` comments before each definition.

I would write each member function in a single line, and add `is_` prefix for the predicates, like this:
```c++
/** @return whether the column has been instantly dropped */
bool is_dropped() const { return data & DROPPED; }
```

Reply:
Fixed except moving to `dict_instant_t`.

Reply:
According to this 5 values (3 bits) are needed:
```c++
	/** The field was unsigned before ALTER to signed:
		0: never
		1: tinyint
		2: smallint
		3: mediumint
		4: int
	*/
```